hydra-node/hydra-node.cabal

```diff
@@ -252,9 +244,7 @@ benchmark tx-cost
 
   ghc-options:    -threaded -rtsopts
 
-  if flag(hydra-development)
-    -- NOTE(SN): should fix HLS choking on PlutusTx plugin
-    ghc-options: -fplugin-opt PlutusTx.Plugin:defer-errors
+  ghc-options: -fplugin-opt PlutusTx.Plugin:defer-errors
```

Comment:
Instead of turning them on for the full package, we should instead reach to module-level disabling of this via `{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:defer-errors #-}` pragmas on the affected modules.

Reply:
Doing this removed the need to do quite some haddock workarounds: a75c0f6765f42015ba9f84a0aea06297feb78daa